report/report.go

```diff
@@ -447,21 +463,21 @@ func (r *Report) CollectCustomMetrics() error {
 }
 
 func (r *Report) CoveragePercent() float64 {
-	if r.Coverage == nil || r.Coverage.Total == 0 {
+	if r == nil || r.Coverage == nil || r.Coverage.Total == 0 {
 		return 0.0
 	}
 	return float64(r.Coverage.Covered) / float64(r.Coverage.Total) * 100
 }
 
 func (r *Report) CodeToTestRatioRatio() float64 {
-	if r.CodeToTestRatio == nil || r.CodeToTestRatio.Code == 0 {
+	if r == nil || r.CodeToTestRatio == nil || r.CodeToTestRatio.Code == 0 {
 		return 0.0
 	}
 	return float64(r.CodeToTestRatio.Test) / float64(r.CodeToTestRatio.Code)
 }
 
 func (r *Report) TestExecutionTimeNano() float64 {
-	if r.TestExecutionTime == nil {
+	if r == nil || r.TestExecutionTime == nil {
```

Comment:
Should I perform a nil check on the interface?

```go
if rPrev != nil || reflect.ValueOf(rPrev).IsNil() == false {
  prev = rPrev.CodeToTestRatioRatio()
}
```

Reply:
I removed the nil check on the caller because I thought it was unnecessary.
https://github.com/k1LoW/octocov/pull/308/commits/7844b9f8010091896f2ffbb47796bc42e9c57ad9